src/lib/CopyBSR.svelte

```diff
@@ -0,0 +1,101 @@
+<script lang="ts">
+  import Fa from 'svelte-fa'
+  import { faTwitch } from '@fortawesome/free-brands-svg-icons'
+  import { onMount } from 'svelte'
+  import { fade } from 'svelte/transition'
+
+  export let fontSize: string = '1rem'
+  export let mapId: string
+
+  // Adding props to allow these to be customizable for the MOTW Section
+  export let tooltipTop: string = '32px'
+  export let tooltipLeft: string = '92%'
+  export let tooltipTransform: string = 'translateX(-50%)'
+
+  // Margins because it looks off on the MapCards without it
+  export let marginRight: string = '4.3px'
+  export let marginTop: string = '3px'
+
+  let showTooltip = false
+  let tooltipTimeout: NodeJS.Timeout
+
+  function copyBSR() {
+    const textToCopy = `!bsr ${mapId}`
+    navigator.clipboard
+      .writeText(textToCopy)
+      .then(() => {
+        showTooltip = true
+        clearTimeout(tooltipTimeout)
+        tooltipTimeout = setTimeout(() => {
+          showTooltip = false
+        }, 3000)
+      })
+      .catch((err) => {
+        console.error('Failed to copy BSR:', err)
+      })
+  }
+
+  onMount(() => {
+    return () => {
+      clearTimeout(tooltipTimeout)
+    }
+  })
+</script>
+
+<button
+  on:click={copyBSR}
+  title="Copy BSR"
+  class="copy-bsr"
+  style="font-size: {fontSize}; margin-right: {marginRight}; margin-top: {marginTop}"
+>
+  <Fa icon={faTwitch} />
+</button>
+
+{#if showTooltip}
+  <div
+    class="tooltip"
+    in:fade={{ duration: 300 }}
```

Comment:
I wouldn't recommend doing the in fade. A click is a pretty immediate action, a more immediate/snappy feedback probably feels better. Otherwise, it could feel rather sluggish and dissociate it from your action. The fade out is okay, imo.

Fading expanding elements on clicks usually feel more snappy since they travel larger distances and make it more clear where the action is happening and "what direction" it's going.